cmd/algorand-indexer/daemon.go

```diff
@@ -107,6 +112,35 @@ var daemonCmd = &cobra.Command{
 	},
 }
 
+// waitForDBAvailable wait for the IndexerDb to report that it is available.
+func waitForDBAvailable(db idb.IndexerDb) {
```

Comment:
I think this is a race between the migration starting and this checking to see if a migration is running

Reply:
This should be OK. If there is going to be a migration it is initialized in the init function BEFORE starting the migration in a separate go routine. The `db.migration` object is initialized with `blocking = true` to ensure there's no race condition where it incorrectly unblocks..